apps/server/src/modules/provisioning/loggable/group-provisioning-info.loggable.ts

```diff
@@ -0,0 +1,27 @@
+import { ErrorLogMessage, Loggable, LogMessage, ValidationErrorLogMessage } from '@src/core/logger';
+import { ExternalGroupDto, ExternalUserDto } from '../dto';
+
+export class GroupProvisioningInfoLoggable implements Loggable {
+	constructor(
+		private readonly groupUser: ExternalUserDto,
+		private readonly groups: ExternalGroupDto[],
+		private readonly durationMs: number
+	) {}
+
+	public getLogMessage(): LogMessage | ErrorLogMessage | ValidationErrorLogMessage {
+		const userCount = this.groups.reduce(
+			(count: number, group: ExternalGroupDto) => count + (group.otherUsers?.length ?? 0),
+			this.groups.length
+		);
+
+		return {
+			message: 'Group provisioning has finished.',
+			data: {
+				externalUserId: this.groupUser.externalId,
```

Comment:
There are two options to entshour the data privacy:
- first shorten the external User Id so that the user if not is traceable
- second option is remove the external User Id